File: problems/two-sum/boilerplate-full/function.js
const fs = require('fs');

const values = fs.readFileSync(0, 'utf8').trim().split(/\s+/).map(Number);
const n = values[0];
const nums = values.slice(1, n + 1);
const target = values[n + 1];
const seen = new Map();

for (let index = 0; index < nums.length; index += 1) {
  const complement = target - nums[index];
  if (seen.has(complement)) {
    console.log(`${seen.get(complement)} ${index}`);
    break;
  }
  seen.set(nums[index], index);
}

File: problems/two-sum/boilerplate-full/function.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

int main() {
	int n;
	std::cin >> n;

	std::vector<int> nums(n);
	for (int& number : nums) {
		std::cin >> number;
	}

	int target;
	std::cin >> target;

	std::unordered_map<int, int> seen;
	for (int index = 0; index < n; ++index) {
		int complement = target - nums[index];
		auto match = seen.find(complement);
		if (match != seen.end()) {
			std::cout << match->second << ' ' << index << '\n';
			return 0;
		}
		seen[nums[index]] = index;
	}

	return 0;
}
